app/controllers/bids_controller.rb

```diff
@@ -31,13 +30,14 @@ def new
   end
 
   def confirm
-    @auction = Presenter::Auction.new(Auction.find(params[:auction_id]))
-    @bid = Presenter::Bid.new(PlaceBid.new(params, current_user).dry_run)
+    @auction = Policy::Auction.factory(Auction.find(params[:auction_id]), current_user)
+    @bid = Policy::Bid.new(PlaceBid.new(@auction, params).dry_run, @auction)
   end
 
   def create
     fail UnauthorizedError, "You must have a valid SAM.gov account to place a bid" unless current_user.sam_account?
-    @bid = Presenter::Bid.new(PlaceBid.new(params, current_user).perform)
+    @auction = Policy::Auction.factory(Auction.find(params[:auction_id]), current_user)
+    @bid = Policy::Bid.new(PlaceBid.new(@auction, params).perform, @auction)
```

Comment:
This seems a little easier to understand, but is complexity that we don't need to see at this level. It could be that the bid policy takes in an auction and params to generate the bid, or .... not sure. Just seems like things are not quite in the right place.